src/App.js

```diff
@@ -134,54 +138,163 @@ class App extends Component {
+      forSetState = this.finishGame(newProgress)
+    } else {
+      forSetState = this.continueGame(newProgress)
+    }
+    this.setState(forSetState)
+  }
+
+  finishGame (progress) {
+    const gameStateObject = Object.assign({}, this.state.game)
+    gameStateObject['gameState'] = WIN
+    const progressObject = Object.assign({}, this.state.progress)
+    return {
+      game: gameStateObject,
+      progress: progressObject
+    }
+  }
+
+  continueGame (progressObj) {
+    let newGameMessage = ''
+    const peopleLiving = this.ignoreStatusDead(this.state.people) // removes all dead characters from play
+    let lessHealthPeopleList = this.peopleLoseHealth(peopleLiving) // decrements all charaters' health, higher decrement if no food
+    let newCharacterList = this.statusDeadChange(lessHealthPeopleList) // changes stauts of one character with health below 0 to "dead"
+    newGameMessage = this.starvedCharacterMessage(newCharacterList) // returns message or empty string
+    let noMessage = newGameMessage.length < 1
+    const newFood = this.decrementFood(newCharacterList)
+    if (this.state.inventory.food > 0 && newFood.food <= 0) { // creates game message when player runs out of food
+      newGameMessage = 'You have run out of food.'
+    }
+    if (noMessage) {
+      const badLuck = randomGenerator(1, 6) === 3
+      const luck = randomGenerator(1, 4)
+      newCharacterList = this.randomCharacterDeath(newCharacterList, badLuck, luck) // returns character list with a depresed or dead character
+      const newlyDepressed = newCharacterList.filter((person) => person.status === 'depressed').filter((person) => person in this.state.people)
+      const newlyDead = newCharacterList.filter((person) => person.status === 'dead').filter((person) => person in this.state.people)
+      const anyNewDepressed = newlyDepressed.length > 0
+      const anyNewlyDead = newlyDead > 0
+      if (anyNewDepressed) {
+        const newlyDepressedName = newlyDepressed.name
+        newGameMessage = newlyDepressedName === 'You' ? `${newlyDepressedName} are depressed.` : `${newlyDepressedName} is depressed.`
+      }
+      if (anyNewlyDead) {
+        const newlyDeadName = newlyDead.name
+        newGameMessage = newlyDeadName === 'You'
+          ? `${newlyDeadName} have died of melancholy.`
+          : `${newlyDeadName} has died of melancholy.`
+      }
     }
-    if (newFood <= 0) {
-      gameMessage = 'You have run out of food.'
-      newFood = 'no food'
+    const youIsDead = this.isYouDead(newCharacterList) // bool
+    const newGameObj = { // gameObj changed for 'You' dead, ends game
+      gameState: youIsDead ? GAMEOVER : PLAYING, // continues or ends game depending on if 'You' in new people list is dead
+      difficulty: this.state.game.difficulty, // preserves difficulty
+      gameMessage: newGameMessage // new game message
+    }
+
+    return {
+      game: newGameObj, // hardcoded from inside function
```

Comment:
I'm going to stop commenting on this now. In general, I believe that comments should only be used as a last resort if your code is unclear of confusing and there is no way to make the code confusing. Look at each comment and ask yourself if it is *really* needed or not. If it is needed, then perhaps you can figure out how to make your code more self-documenting.